File: 백준/Bronze/2909. 캔디 구매/캔디 구매.c
#include <stdio.h>
#include <math.h>

int main() {
    int C, K;
    scanf("%d %d", &C, &K);
    int t = C % (int)pow(10, K);
    if (t >= pow(10, K)/2){
        C += (int)pow(10, K);
    }
    C -= t;
    printf("%d\n", C);
    return 0;
}

File: 백준/Bronze/2909. 캔디 구매/캔디 구매.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int C = sc.nextInt();
        int K = sc.nextInt();
        int t = C % (int) Math.pow(10, K);
        if (t >= Math.pow(10, K)/2) {
            C += (int) Math.pow(10, K);
        }
        C -= t;
        System.out.println(C);
        sc.close();
    }
}